### ISSUE TITLE
`pip install -e .` fails at `node scripts/dedupe.js`

### ISSUE THREAD

**Issue by gnestor** [jupyter/jupyterlab#1057]
It looks like it fails at `node scripts/dedupe.js` with "Cannot resolve module 'jupyterlab/lib/editorhandler/plugin' in /Users/grant/Sites/jupyter/jupyterlab/jupyterlab". It's looking for `jupyterlab/jupyterlab/lib/editorhandler/plugin` which doesn't exist but `jupyterlab/lib/editorhandler/plugin` does.

```
Obtaining file:///Users/grant/Sites/jupyter/jupyterlab
    Complete output from command python setup.py egg_info:
    setup.py entered
    ...
    
    > jupyterlab@0.0.5 prepublish /Users/grant/Sites/jupyter/jupyterlab
    > npm run build
    
    
    > jupyterlab@0.0.5 build /Users/grant/Sites/jupyter/jupyterlab
    > npm run build:src
    
    
    > jupyterlab@0.0.5 build:src /Users/grant/Sites/jupyter/jupyterlab
    > tsc --project src && node scripts/copyfiles.js
    
    npm WARN prefer global marked@0.3.6 should be installed with -g
    
    > fsevents@1.0.14 install /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/fsevents
    > node-pre-gyp install --fallback-to-build
    
    [fsevents] Success: "/Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/fsevents/lib/binding/Release/node-v48-darwin-x64/fse.node" is installed via remote
    
    > phosphor-dragdrop@0.9.1 postinstall /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/phosphor-dragdrop
    > npm dedupe
    
    
    > phosphor-keymap@0.8.0 postinstall /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/phosphor-keymap
    > npm dedupe
    
    
    > phosphor-widget@1.0.0-rc.1 postinstall /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/phosphor-widget
    > npm dedupe
    
    
    > phosphor-codemirror@0.0.1 postinstall /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/phosphor-codemirror
    > npm dedupe
    
    
    > phosphor-commandpalette@0.2.0 postinstall /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/phosphor-commandpalette
    > npm dedupe
    
    
    > phosphor-menus@1.0.0-rc.1 postinstall /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/phosphor-menus
    > npm dedupe
    
    
    > phosphor-panel@1.0.0-rc.1 postinstall /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/phosphor-panel
    > npm dedupe
    
    
    > phosphor-boxpanel@1.0.0-rc.0 postinstall /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/phosphor-boxpanel
    > npm dedupe
    
    
    > phosphor-splitpanel@1.0.0-rc.1 postinstall /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/phosphor-splitpanel
    > npm dedupe
    
    
    > phosphor-stackedpanel@1.0.0-rc.0 postinstall /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/phosphor-stackedpanel
    > npm dedupe
    
    
    > phosphor-tabs@1.0.0-rc.2 postinstall /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/phosphor-tabs
    > npm dedupe
    
    
    > phosphor-dockpanel@0.9.7 postinstall /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/phosphor-dockpanel
    > npm dedupe
    
    
    > phosphide@0.9.5 postinstall /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/phosphide
    > npm dedupe
    
    
    > jupyterlab@0.0.5 postinstall /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/jupyterlab
    > npm dedupe
    
    
    > jupyterlab-extension@0.0.1 postinstall /Users/grant/Sites/jupyter/jupyterlab/jupyterlab
    > npm dedupe
    
    ...
    
    > jupyterlab-extension@0.0.1 build /Users/grant/Sites/jupyter/jupyterlab/jupyterlab
    > npm update jupyterlab && webpack --config webpack.conf.js
    
    
    > jupyterlab@0.0.5 prepublish /Users/grant/Sites/jupyter/jupyterlab
    > npm run build
    
    
    > jupyterlab@0.0.5 build /Users/grant/Sites/jupyter/jupyterlab
    > npm run build:src
    
    
    > jupyterlab@0.0.5 build:src /Users/grant/Sites/jupyter/jupyterlab
    > tsc --project src && node scripts/copyfiles.js
    
    
    > jupyterlab@0.5.0 postinstall /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/jupyterlab
    > node scripts/dedupe.js
    
    - bootstrap@3.3.7 node_modules/bootstrap
    - d3-format@0.5.1 node_modules/d3-format
    - jquery-ui@1.12.1 node_modules/jquery-ui
    - jupyter-js-utils@0.4.0 node_modules/jupyter-js-utils
    - backbone@1.2.0 node_modules/jupyter-js-widgets/node_modules/backbone
    - lolex@1.5.1 node_modules/lolex
    - phosphor-codemirror@0.0.1 node_modules/phosphor-codemirror
    - phosphor-menus@1.0.0-rc.1 node_modules/phosphor-menus
    - phosphor-observablelist@1.0.0-beta node_modules/phosphor-observablelist
    - sanitizer@0.1.3 node_modules/sanitizer
    - scriptjs@2.5.8 node_modules/scriptjs
    - jupyter-js-widgets@2.0.0-dev.0 node_modules/jupyter-js-widgets
    jupyterlab-extension@0.0.1 /Users/grant/Sites/jupyter/jupyterlab/jupyterlab
    ├── jupyterlab@0.5.0  invalid
    └── webpack@1.13.2
    
    ModuleNotFoundError: Module not found: Error: Cannot resolve module 'jupyterlab/lib/editorhandler/plugin' in /Users/grant/Sites/jupyter/jupyterlab/jupyterlab
        at /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/webpack/lib/Compilation.js:229:38
        at onDoneResolving (/Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/webpack/lib/NormalModuleFactory.js:29:20)
        at /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/webpack/lib/NormalModuleFactory.js:85:20
        at /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/async/lib/async.js:726:13
        at /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/async/lib/async.js:52:16
        at done (/Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/async/lib/async.js:241:17)
        at /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/async/lib/async.js:44:16
        at /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/async/lib/async.js:723:17
        at /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/async/lib/async.js:167:37
        at /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/enhanced-resolve/lib/UnsafeCachePlugin.js:24:19
        at onResolved (/Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/enhanced-resolve/lib/Resolver.js:38:18)
        at innerCallback (/Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/enhanced-resolve/lib/Resolver.js:94:11)
        at loggingCallbackWrapper (/Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/enhanced-resolve/lib/createInnerCallback.js:21:19)
        at /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/tapable/lib/Tapable.js:134:6
        at /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/enhanced-resolve/lib/ModulesInDirectoriesPlugin.js:54:23
        at /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/enhanced-resolve/lib/Resolver.js:191:15
        at /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/enhanced-resolve/lib/ModulesInDirectoriesPlugin.js:45:26
        at loggingCallbackWrapper (/Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/enhanced-resolve/lib/createInnerCallback.js:21:19)
        at /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/tapable/lib/Tapable.js:134:6
        at /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/enhanced-resolve/lib/Resolver.js:122:33
        at /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/enhanced-resolve/lib/Resolver.js:191:15
        at applyPluginsParallelBailResult.createInnerCallback.log (/Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/enhanced-resolve/lib/Resolver.js:110:4)
        at loggingCallbackWrapper (/Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/enhanced-resolve/lib/createInnerCallback.js:21:19)
        at /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/tapable/lib/Tapable.js:134:6
        at Tapable.<anonymous> (/Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/enhanced-resolve/lib/DirectoryDescriptionFilePlugin.js:24:12)
        at Storage.finished (/Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:38:16)
        at ReadFileContext.callback (/Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/graceful-fs/graceful-fs.js:78:16)
        at FSReqWrap.readFileAfterOpen [as oncomplete] (fs.js:365:13)
    resolve module jupyterlab/lib/editorhandler/plugin in /Users/grant/Sites/jupyter/jupyterlab/jupyterlab
      looking for modules in /Users/grant/Sites/jupyter/jupyterlab/node_modules
        /Users/grant/Sites/jupyter/jupyterlab/node_modules/jupyterlab doesn't exist (module as directory)
      looking for modules in /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules
        resolve 'file' or 'directory' lib/editorhandler/plugin in /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/jupyterlab
          resolve file
            /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/jupyterlab/lib/editorhandler/plugin doesn't exist
            /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/jupyterlab/lib/editorhandler/plugin.webpack.js doesn't exist
            /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/jupyterlab/lib/editorhandler/plugin.web.js doesn't exist
            /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/jupyterlab/lib/editorhandler/plugin.js doesn't exist
            /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/jupyterlab/lib/editorhandler/plugin.json doesn't exist
          resolve directory
            /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/jupyterlab/lib/editorhandler/plugin doesn't exist (directory default file)
            /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/node_modules/jupyterlab/lib/editorhandler/plugin/package.json doesn't exist (directory description file)
    
    npm ERR! Darwin 16.0.0
    npm ERR! argv "/usr/local/bin/iojs" "/usr/local/bin/npm" "run" "build"
    npm ERR! node v6.3.0
    npm ERR! npm  v3.10.3
    npm ERR! code ELIFECYCLE
    npm ERR! jupyterlab-extension@0.0.1 build: `npm update jupyterlab && webpack --config webpack.conf.js`
    npm ERR! Exit status 1
    npm ERR!
    npm ERR! Failed at the jupyterlab-extension@0.0.1 build script 'npm update jupyterlab && webpack --config webpack.conf.js'.
    npm ERR! Make sure you have the latest version of node.js and npm installed.
    npm ERR! If you do, this is most likely a problem with the jupyterlab-extension package,
    npm ERR! not with npm itself.
    npm ERR! Tell the author that this fails on your system:
    npm ERR!     npm update jupyterlab && webpack --config webpack.conf.js
    npm ERR! You can get information on how to open an issue for this project with:
    npm ERR!     npm bugs jupyterlab-extension
    npm ERR! Or if that isn't available, you can get their info via:
    npm ERR!     npm owner ls jupyterlab-extension
    npm ERR! There is likely additional logging output above.
    
    npm ERR! Please include the following file with any support request:
    npm ERR!     /Users/grant/Sites/jupyter/jupyterlab/jupyterlab/npm-debug.log
    /Users/grant/anaconda/lib/python3.5/distutils/dist.py:261: UserWarning: Unknown distribution option: 'License'
      warnings.warn(msg)
    rebuilding js and css failed
    missing files: ['/Users/grant/Sites/jupyter/jupyterlab/jupyterlab/build/bundle.js']
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/Users/grant/Sites/jupyter/jupyterlab/setup.py", line 161, in <module>
        setup(**setup_args)
      File "/Users/grant/anaconda/lib/python3.5/distutils/core.py", line 148, in setup
        dist.run_commands()
      File "/Users/grant/anaconda/lib/python3.5/distutils/dist.py", line 955, in run_commands
        self.run_command(cmd)
      File "/Users/grant/anaconda/lib/python3.5/distutils/dist.py", line 974, in run_command
        cmd_obj.run()
      File "/Users/grant/Sites/jupyter/jupyterlab/setup.py", line 51, in run
        raise e
      File "/Users/grant/Sites/jupyter/jupyterlab/setup.py", line 44, in run
        self.distribution.run_command('jsdeps')
      File "/Users/grant/anaconda/lib/python3.5/distutils/dist.py", line 974, in run_command
        cmd_obj.run()
      File "/Users/grant/Sites/jupyter/jupyterlab/setup.py", line 109, in run
        check_call(['npm', 'run', 'build'], cwd=node_root, stdout=sys.stdout, stderr=sys.stderr)
      File "/Users/grant/anaconda/lib/python3.5/subprocess.py", line 581, in check_call
        raise CalledProcessError(retcode, cmd)
    subprocess.CalledProcessError: Command '['npm', 'run', 'build']' returned non-zero exit status 1
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /Users/grant/Sites/jupyter/jupyterlab/
```

**Comment by blink1073** [comment]
It looks like you need to run `git clean -dfx`, we're not using any of the `phosphor-*` repos anymore.

**Comment by gnestor** [comment]
Unfortunately I tried that... I tried with Node 5 and 6. I even tried uninstalling my local and installing the latest release from pip and I get the same error. The error message is pretty informative (`Module not found: Error: Cannot resolve module 'jupyterlab/lib/editorhandler/plugin' in /Users/grant/Sites/jupyter/jupyterlab/jupyterlab`). It looks like it's webpack that's trying to resolve the module, but I'm not sure what is the culprit in the webpack config...
